### PR TITLE
Format numbers using numeraljs

### LINKED ISSUE #198: Internationalize and humanize numbers (at the same time)
# 🙋 Feature Request

Now that the numbers are displayed in a human-readable form, would be nice to also make them internationalized.

## 😯 Describe the feature

It is hard to describe all the requirements specifically, without knowing the languages. This should be covered by a library.

## 💻 Examples

Here is an non-exhaustive list of requirements:
 - [ ] in some locales comma is used instead of the dot as a decimal separator
 - [ ] in some locales thousands (or even tenth of thousands) are space-separated
 - [ ] exponent suffixes "k", "M"  should ideally be translated, (e.g. "k" is "Tnd." in German and "тыс." in Russian)

## 💁 Possible Solution

A mature library should be used. Either D3.js  currently used or number formatting, or react-i18next used for i18n or both.

## Related:

 - This is a continuation of https://github.com/neherlab/covid19_scenarios/issues/158  implemented in https://github.com/neherlab/covid19_scenarios/pull/196 (except for i18n part)
 - https://github.com/neherlab/covid19_scenarios/issues/56

### PR DESCRIPTION
**Pull request by rcbevans**
Related issues and PRs
Alternative to #213
Fixes #198

Description
Use numeraljs to allow internationalizing and human formatting values simultaneously

Screenshot of spanish language with humanized formatting

Impacted Areas in the application
Result charts and tables.

Testing
Switch language with a scernario ran, and humaized results enabled and validate that axis, tooltips and tables are showing appropriately humanized, and internationalized values.

### CODE REVIEW

File: src/helpers/numberFormat.ts [hunk 1]
```diff
@@ -1,6 +1,5 @@
-export function numberFormatter(language: string | undefined, humanize: boolean, round: boolean) {
-  return new Intl.NumberFormat(language, {
-    ...(humanize ? { notation: 'compact', compactDisplay: 'short' } : {}),
-    ...(round ? { maximumFractionDigits: 0 } : {}),
-  }).format
+import { numeral } from '../i18n'
+
+export function numberFormatter(humanize: boolean, round: boolean) {
+  return (value: number) => numeral(value).format(`0${!round ? '.[000]' : ''}${humanize ? 'a' : ''}`)
```
Comment (gj262) [inline src/helpers/numberFormat.ts]: Hi Rich, For non humanize to get 1000000 -> 1,000,000, I think a prefix `0,` is required. i.e. `format(`0,0${!roun`...
Screen Shot 2020-03-26 at 4 23 04 PM

Comment (ivan-aksamentov) [reply src/helpers/numberFormat.ts]: @gj262 @rcbevans Strangely, this library gives small "m" for millions in English and German, but capital "M" for French. I don't get it. Looks like "meters" to me.

**Comment by ivan-aksamentov** [comment]
@gj262 @rcbevans Folks, I've merged like 256 issues this week and now my brain is completely toasted. Could you please remind me why we are still doing this?

According to my Firefox and Chrome everything works smoothly, and suffixes are prettily internationalized. Was there an issue with too much rounding, right?

**Comment by rcbevans** [comment]
On my devices numbers are not abbreviated, they are just showing grouped digits.

 Tested in Firefox on Android, Firefox and Safari on Mac.

Chrome is the only browser which is correctly abbreviating values.

**Comment by rcbevans** [comment]
On chrome where values are abbreviated, they are rounded to integers, which, for very large values in the millions can be a very large displayed margin of error, i.e 1.67M was showing as 2M.

**Comment by gj262** [comment]
I think the tl;dr; for the toasted brain is merge PR #241. 

It fixes most of the issues for #198 and numeralJs is configurable if we want to go further.

CC @ivan-aksamentov @rcbevans

**Comment by ivan-aksamentov** [comment]
@rcbevans @gj262 There is a problem with `numeral` on Polish locale.
I reopened https://github.com/neherlab/covid19_scenarios/issues/158
Could you please take a look?

**Comment by rcbevans** [comment]
> @rcbevans @gj262 There is a problem with `numeral` on Polish locale.
> I reopened #158
> Could you please take a look?

The never ending number humanization/internationalization saga.  How can this be so hard 😒

**Comment by ivan-aksamentov** [comment]
@rcbevans We've got some of your colleagues from MS helping us. Would you like to join our casual Slack channel? Send me your email on Spectrum if you do.

**Comment by rcbevans** [comment]
> @rcbevans We've got some of your colleagues from MS helping us. Would you like to join our casual Slack channel? Send me your email on Spectrum if you do.

Slack, not teams!? 😱

😂

**Pull request merged**